### ISSUE TITLE
Session Aggregates are randomly dropped

### ISSUE THREAD

**Issue by Dav1dde** [getsentry/relay#2967]
Session aggregates are being dropped/lost before the reach Kafka/Storage.

We could verify by peeking into the Kafka Topic that the aggregates (as metrics) never make it into Kafka.

From the Kafka Topic: 

```
 (datetime.datetime(2024, 1, 18, 10, 9), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 11), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 12), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 14), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 17), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 19), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 21), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 23), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 25), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 29), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 31), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 32), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 34), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 35), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 40), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 49), 30.0),
 (datetime.datetime(2024, 1, 18, 10, 56), 30.0)]
```

In the UI:
![image](https://github.com/getsentry/relay/assets/255721/fae8b100-bd1d-4600-8a46-2df25218953d)

Expected to see one entry every minute with a value of 30.

Script which generates an envelop containing a session aggregate every minute:

```sh
#!/usr/bin/env bash


URL="https://<INGEST URL>/api/<PROJECT ID>/envelope/?sentry_key=<KEY>&sentry_version=7&sentry_client=sentry.javascript.browser%2F7.93.0"

RELEASE="r1"
ENVIRONMENT="production"
USER_AGENT="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"

while true; do
    DT=$(date -u +'%Y-%m-%dT%H:%M:%S.398Z')
    echo $DT

    PAYLOAD="{\"sent_at\":\"${DT}\",\"sdk\":{\"name\":\"sentry.javascript.browser\",\"version\":\"7.93.0\"}}
{\"type\":\"sessions\"}
{\"attrs\":{\"release\":\"$RELEASE\", \"environment\":\"$ENVIRONMENT\", \"user_agent\":\"$USER_AGENT\"},\"aggregates\":[{\"started\":\"$(date -u +'%Y-%m-%dT%H:%M:00.000Z')\",\"exited\":100}]}"

    echo $PAYLOAD 
    curl -H 'User-Agent:' -H 'Accept:' -H "Content-Type: application/x-sentry-envelope" -X POST "$URL" -d "$PAYLOAD"

    echo ""
    sleep 60
done
```

I was not able to reproduce this with a local Sentry and Relay instance, Relay consistently delivered the metrics into the Kafka topic.

The local instance had the following non-standard settings:

```py
SENTRY_FEATURES['organizations:metrics-extraction'] = True
SENTRY_FEATURES['organizations:transaction-metrics-extraction'] = True
SENTRY_FEATURES['organizations:custom-metrics'] = True
SENTRY_FEATURES['organizations:ddm-ui'] = True
SENTRY_FEATURES['organizations:release-health-drop-sessions'] = True

SENTRY_RELEASE_HEALTH = "sentry.release_health.metrics.MetricsReleaseHealthBackend"
SENTRY_RELEASE_MONITOR = "sentry.release_health.release_monitor.metrics.MetricReleaseMonitorBackend"
```

I manually verified that the `sessionMetrics` setting in the project config matches the production one `version: 2, drop: true`.

**Comment by Dav1dde** [comment]
Ingesting the session aggregates directly into processing relays (skipping pops) with the same above script does not show this issue (using a value of 100 instead of 30):

<img width="1191" alt="image" src="https://github.com/getsentry/relay/assets/255721/cc7e5287-823b-429d-a598-e6ae6cd6fd3a">

The test was performed by sending the payload to a single Relay instance everytime, this does not rule out that there may be a problem with non-cached Project Configs.

**Comment by Dav1dde** [comment]
Still no issues when running it against random processing relays (should rule out issues with non-cached project configs):

<img width="1206" alt="image" src="https://github.com/getsentry/relay/assets/255721/7a078ee3-8afa-412e-8e0f-9b1915b3443e">

**Comment by Dav1dde** [comment]
It appears like running the script against a single pop instance also works.

**Comment by Dav1dde** [comment]
Running the script against random pop instances shows the same behaviour:

<img width="1212" alt="image" src="https://github.com/getsentry/relay/assets/255721/e1492818-c942-46ce-9de0-976d9a154870">

**Comment by jjbayer** [comment]
Did not notice anything obvious in the code (`session aggregate -> aggregation -> submission via global metrics endpoint`), will try to repro the behavior with an integration test now.

**Comment by jjbayer** [comment]
There's definitely an edge case in which we aggregate & attempt to send metrics even though the project has expired (see https://github.com/getsentry/relay/pull/2987), but I doubt that it explains the behavior above.

**Comment by jjbayer** [comment]
Seems fixed by #2987:

![image](https://github.com/getsentry/relay/assets/1565449/22a7b56f-c03e-4d65-b107-43f6115be30c)

Problem seems to have been project configs not being actively fetched when the only incoming traffic (from processing relay's point of view) is metrics via the global endpoint. See linked PR for details.

**Comment by Dav1dde** [comment]
Waiting for customer to confirm that the issue is fixed for them.
